vendor/github.com/swiftstack/sortedmap/llrb_tree.go

```diff
@@ -36,6 +36,12 @@ type llrbTreeStruct struct {
 
 // API functions (see api.go)
 
+// Return the tree to its original, empty, state.
+//
+func (tree *llrbTreeStruct) Reset() {
+	tree.root = nil
```

Comment:
That should do it... although simply dropping the reference to the LLRBTree itself would accomplish the same thing. I'm going to presume that you found a case where you want to re-use an LLRBTree?

Reply:
llrbTreeStruct has an embedded Mutex. I was trying to cut down on the instances where we allocate a Mutex and then discard it (for garbage collection to clean up) because I want to cut down on the number of "active" Mutex that the trackedlock lock watcher has to keep track of (the lock watcher gives up on Mutex and RWMutex that have not been used for 20 sec (configurable), so it will continue to hold a reference to these Mutex for 20 sec).  From both the GC perspective and the lock watcher perspective, its better to reuse the Mutex via a Pool then leave them lying around for 20 sec and then letting the GC clean them up.

Reply:
I get it... the overhead of creating/destroying an LLRB tree is likely dominated by the Mutex... assuming the tree is relatively empty.